packages/plugin/src/lib/paystack.ts

```diff
@@ -40,48 +41,76 @@ export default class Paystack {
     this.apiKey = apiKey;
   }
 
+  // protected async requestPaystackAPI<T>(request: Request): Promise<T> {
+  //   const path =
+  //     request.path.replace(/\/$/, "") +
+  //     "/?" +
+  //     new URLSearchParams(request.query).toString();
+
+  //   const options = {
+  //     method: request.method,
+  //     path,
+  //     headers: {
+  //       Authorization: `Bearer ${this.apiKey}`,
+  //       "Content-Type": "application/json",
+  //     },
+  //   };
+
+  //   return new Promise((resolve, reject) => {
+  //     const req = https.request(PAYSTACK_API_PATH, options, res => {
+  //       let data: Uint8Array[] = [];
+
+  //       res.on("data", chunk => {
+  //         data.push(chunk);
+  //       });
+
+  //       res.on("end", () => {
+  //         try {
+  //           resolve(JSON.parse(Buffer.concat(data).toString()) as T);
+  //         } catch (e) {
+  //           reject(e);
+  //         }
+  //       });
+  //     });
+
+  //     req.on("error", e => {
+  //       reject(e);
+  //     });
+
+  //     if (request.body && Object.values(request.body).length > 0) {
+  //       req.write(JSON.stringify(request.body));
+  //     }
+
+  //     req.end();
+  //   });
+  // }
+
   protected async requestPaystackAPI<T>(request: Request): Promise<T> {
     const path =
       request.path.replace(/\/$/, "") +
       "/?" +
       new URLSearchParams(request.query).toString();
+    let data;
 
     const options = {
       method: request.method,
-      path,
+      url: PAYSTACK_API_PATH + path,
       headers: {
         Authorization: `Bearer ${this.apiKey}`,
         "Content-Type": "application/json",
       },
     };
 
-    return new Promise((resolve, reject) => {
-      const req = https.request(PAYSTACK_API_PATH, options, res => {
-        let data: Uint8Array[] = [];
-
-        res.on("data", chunk => {
-          data.push(chunk);
-        });
-
-        res.on("end", () => {
-          try {
-            resolve(JSON.parse(Buffer.concat(data).toString()) as T);
-          } catch (e) {
-            reject(e);
-          }
-        });
-      });
-
-      req.on("error", e => {
-        reject(e);
-      });
-
-      if (request.body && Object.values(request.body).length > 0) {
-        req.write(JSON.stringify(request.body));
-      }
-
-      req.end();
-    });
+    try {
+      const res = await axios(
+        request.body ? { ...options, data: request.body } : { ...options }
+      );
+      data = res.data;
+      return res.data;
+    } catch (error) {
+      throw new Error("something went wrong");
+    }
+    return data;
```

Comment:
This return is redundant, execution will never get here, you should remove this return and the `data` assignment on [line 108](https://github.com/a11rew/medusa-payment-paystack/pull/52/files#diff-70aff2816e85d05d93473c91f0e93e464dc03a4ddac8bbb7957e159470966f09R108) too.